apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx

```diff
@@ -1,33 +1,110 @@
-import type {
-  InvalidValue,
-  LayersValue,
-  TupleValue,
+import {
+  toValue,
+  type InvalidValue,
+  type LayersValue,
+  type RgbValue,
+  type StyleValue,
+  type TupleValue,
+  type UnitValue,
 } from "@webstudio-is/css-engine";
-import { parseBoxShadow } from "@webstudio-is/css-data";
+import {
+  extractBoxShadowProperties,
+  parseBoxShadow,
+  type ExtractedBoxShadowProperties,
+} from "@webstudio-is/css-data";
 import {
   Flex,
+  Grid,
   Label,
+  Separator,
   Text,
   TextArea,
   textVariants,
   theme,
+  ToggleGroup,
+  ToggleGroupButton,
   Tooltip,
 } from "@webstudio-is/design-system";
-import { InformationIcon } from "@webstudio-is/icons";
-import { useState } from "react";
+import {
+  InformationIcon,
+  ShadowInsetIcon,
+  ShadowNormalIcon,
+} from "@webstudio-is/icons";
+import { useMemo, useState } from "react";
 import type { IntermediateStyleValue } from "../../shared/css-value-input";
+import { CssValueInputContainer } from "../../controls/position/css-value-input-container";
+import { toPascalCase } from "../../shared/keyword-utils";
+import { ColorControl } from "../../controls";
+import type { SetProperty } from "../../shared/use-style-data";
+
+/*
+  We can't split and use individual css properties for box-shadow.
+  So, the best way to get eh validation from inputs is to use a fallback
```

Comment:
what's `eh`?

Reply:
Just a typo, going to correct the comment 👍 😅